pkg/pillar/cmd/zedrouter/cni.go

```diff
@@ -0,0 +1,431 @@
+		return err
+	}
+	_, err = z.niReconciler.UpdateAppConn(
+		z.runCtx, *appConfig, appStatus.AppPod, vifs)
+	if err != nil {
+		z.log.Error(err)
+		return err
+	}
+	return nil
+}
+
+func (z *zedrouter) handleCheckPodConnectionRequest(
+	args cnirpc.CheckPodConnectionArgs, retval *cnirpc.CheckPodConnectionRetval) error {
+	_, appStatus, err := z.getAppByPodName(args.Pod.Name)
+	if err != nil {
+		z.log.Error(err)
+		return err
+	}
+	var ulStatus *types.UnderlayNetworkStatus
+	for i := range appStatus.UnderlayNetworkList {
+		ulStatus = &appStatus.UnderlayNetworkList[i]
+		if ulStatus.PodVif.GuestIfName == args.PodInterface.Name {
+			break
+		}
+		ulStatus = nil
+	}
+	if ulStatus == nil {
+		err = fmt.Errorf("failed to find adapter with pod interface %s for app %v",
+			args.PodInterface.Name, appStatus.UUIDandVersion.UUID)
+		z.log.Error(err)
+		return err
+	}
+	// Find out if DHCP is being used.
+	netInstStatus := z.lookupNetworkInstanceStatus(ulStatus.Network.String())
+	if netInstStatus == nil {
+		// Should be unreachable.
+		err = fmt.Errorf("missing network instance status for %s",
+			ulStatus.Network.String())
+		z.log.Error(err)
+		return err
+	}
+	retval.UsesDHCP = z.niWithDHCP(netInstStatus)
+	// Check the state of the connection configuration in the network stack.
+	appConnRecStatus, err := z.niReconciler.GetAppConnStatus(appStatus.UUIDandVersion.UUID)
+	if err != nil {
+		z.log.Error(err)
+		return err
+	}
+	var vifStatus *nireconciler.AppVIFReconcileStatus
+	for _, vif := range appConnRecStatus.VIFs {
+		if vif.NetAdapterName == ulStatus.Name {
+			vifStatus = &vif
+			break
+		}
+	}
+	if vifStatus == nil {
+		err = fmt.Errorf("missing VIF status for adapter %s", ulStatus.Name)
+		z.log.Error(err)
+		return err
+	}
+	if vifStatus.InProgress {
+		// It is not expected that some config items are created asynchronously.
+		err = fmt.Errorf("some config items related to VIF %v/%s are still in progress",
+			appStatus.UUIDandVersion.UUID, ulStatus.Name)
+		z.log.Error(err)
+		return err
+	}
+	var failedItems []string
+	for itemRef, itemErr := range vifStatus.FailedItems {
+		failedItems = append(failedItems, fmt.Sprintf("%v (%v)", itemRef, itemErr))
+	}
+	if len(failedItems) > 0 {
+		err = fmt.Errorf("failed config items: %s", strings.Join(failedItems, ";"))
+		z.log.Error(err)
+		return err
+	}
+	return nil
+}
+
+func (z *zedrouter) niWithDHCP(netInstStatus *types.NetworkInstanceStatus) bool {
+	airGapped := netInstStatus.PortLogicalLabel == ""
+	switchNI := netInstStatus.Type == types.NetworkInstanceTypeSwitch
+	return !switchNI || !airGapped
+}
+
+func (z *zedrouter) getAppByPodName(
+	podName string) (*types.AppNetworkConfig, *types.AppNetworkStatus, error) {
+	appName, appUUIDPrefix, err := kubeapi.GetAppNameFromPodName(podName)
+	if err != nil {
+		return nil, nil, err
+	}
+	for _, item := range z.pubAppNetworkStatus.GetAll() {
+		appStatus := item.(types.AppNetworkStatus)
+		if appStatus.DisplayName == appName &&
```

Comment:
Need to convert the 'DisplayName' to lower case to match in this case

Reply:
Done in https://github.com/naiming-zededa/eve/pull/230